### DISCUSSION TITLE
Issuance time for products in issuing office time zone

### DISCUSSION THREAD

**Discussion by JJKraw** [weather-gov/api#604] [Q&A]
What I'm trying to do: When I pull the index of a particular product for a particular office, e.g.

  https://api.weather.gov/products/types/LSR/locations/LOX

I'd like to display the issuanceTimes using the time zone of the office (in this case, Pacific Standard Time for Los Angeles). However, other than scraping the text of one of the products for a time zone hint ("PST"), I can't find the time zone of the office within the API.

https://api.weather.gov/offices/{officeID} does not have a time zone field. I can go into a responsible county/zone where there _is_ a time zone field, but that doesn't work if an office is responsible for locations in multiple time zones and I pull a county/zone in a zone different from the office.

Each product itself has an issuanceTime both in the index and the product, but those all appear to be in GMT.

So... Am I totally missing something?? Any ideas if not? My thought is to just cook up my own data to reference to get office ID -> time zone.

Thanks!!

**Comment by sullynole** [comment]
Some offices have more than one time zone within their county warning area so depending on what you are trying to do it may not always make sense to match the time zone to the office.  I think any local time zone would have to be scraped from the productText field outside of using a location to timezone match.  Generally for watch/warning and forecast text products the text header will be the local time zone of the affected area and not necessarily the local office, but there may be nuances for some products such as LSRs.

**Reply by akrherz** [reply]
Oh, the lovely edge cases, as you suggest:

```
SXKA80 PTKK 110600
OMRTKK
SURF OBSERVATIONS
NATIONAL WEATHER SERVICE CHUUK FM
11 PM LST SUN DEC 11 2022
```

```
FLUS43 KLOT 071011
HWOLOT

Hazardous Weather Outlook
National Weather Service Chicago/Romeoville IL
411 AM CST Tue Feb 7 2017

ILZ003>006-008-010>014-019>023-032-033-039-INZ001-002-010-011-019-
081015-
Winnebago-Boone-McHenry-Lake Illinois-Ogle-Lee-De Kalb-Kane-
DuPage-Cook-La Salle-Kendall-Grundy-Will-Kankakee-Livingston-
Iroquois-Ford-Lake Indiana-Porter-Newton-Jasper-Benton-
411 AM CST Tue Feb 7 2017 /511 AM EST Tue Feb 7 2017/
```

**Comment by scadergit** [comment]
We'll add it if you get the house to pass the savings time bill approved by senate last March. =)

**Reply by scadergit** [reply]
If implemented, it would likely be the IANA timezones, like America/Los_Angeles. Thoughts?

**Comment by JJKraw** [comment]
That makes total sense as it is consistent with what is contained within the zones data.

**Comment by JJKraw** [comment]
I came up with a kludgy workaround to get what I need for now.

1) I have a zip code to coordinates piece that I use for forecast look ups, and, for most office locations, there is a zip code in the API "offices" endpoint. I can map the zip code to a point and then do an API points call with the lat/long I get back, and THAT has a timezone within it.

2) I already cache some info for offices (refreshed every two week via cron job) so I just added the timezone to that cache.

3) Special cases: AER and ALU (just use result from Anchorage); BVT (no zip code for Burlington for some reason), PPG (returning a 404 for me right now).

Result below if anyone is interested...

  ABQ  Albuquerque, NM  America/Denver
  ABR  Aberdeen, SD  America/Chicago
  AER  Anchorage East  America/Anchorage
  AFC  Anchorage, AK  America/Anchorage
  AFG  Fairbanks, AK  America/Anchorage
  AJK  Juneau, AK  America/Anchorage
  AKQ  Wakefield, VA  America/New_York
  ALU  Anchorage West  America/Anchorage
  ALY  Albany, NY  America/New_York
  AMA  Amarillo, TX  America/Chicago
  APX  Gaylord, MI  America/Detroit
  ARX  La Crosse, WI  America/Chicago
  BGM  Binghamton, NY  America/New_York
  BIS  Bismarck, ND  America/Chicago
  BMX  NWS Birmingham, Alabama  America/Chicago
  BOI  Boise, ID  America/Boise
  BOU  Denver/Boulder, CO  America/Denver
  BOX  Boston / Norton, MA  America/New_York
  BRO  Brownsville/Rio Grande Valley, TX  America/Chicago
  BTV  Burlington, VT  America/New_York
  BUF  Buffalo, NY  America/New_York
  BYZ  Billings, MT  America/Denver
  CAE  Columbia, SC  America/New_York
  CAR  Caribou, ME  America/New_York
  CHS  Charleston, SC  America/New_York
  CLE  Cleveland, OH  America/New_York
  CRP  Corpus Christi, TX  America/Chicago
  CTP  State College, PA  America/New_York
  CYS  Cheyenne, WY  America/Denver
  DDC  Dodge City, KS  America/Chicago
  DLH  Duluth, MN  America/Chicago
  DMX  Des Moines, IA  America/Chicago
  DTX  Detroit/Pontiac, MI  America/Detroit
  DVN  Quad Cities, IA/IL  America/Chicago
  EAX  Kansas City/Pleasant Hill, MO  America/Chicago
  EKA  Eureka, CA  America/Los_Angeles
  EPZ  El Paso, TX  America/Denver
  EWX  Austin/San Antonio, TX  America/Chicago
  FFC  Peachtree City, GA  America/New_York
  FGF  Grand Forks, ND  America/Chicago
  FGZ  Flagstaff, AZ  America/Phoenix
  FSD  Sioux Falls, SD  America/Chicago
  FWD  Fort Worth/Dallas, TX  America/Chicago
  GGW  Glasgow, MT  America/Denver
  GID  Hastings, NE  America/Chicago
  GJT  Grand Junction, CO  America/Denver
  GLD  Goodland, KS  America/Denver
  GRB  Green Bay, WI  America/Chicago
  GRR  Grand Rapids, MI  America/Detroit
  GSP  Greenville-Spartanburg, SC  America/New_York
  GUM  Tiyan, GU  Pacific/Guam
  GYX  Gray - Portland, ME  America/New_York
  HFO  Honolulu, HI  Pacific/Honolulu
  HGX  Houston/Galveston, TX  America/Chicago
  HNX  San Joaquin Valley, CA  America/Los_Angeles
  HUN  Huntsville, AL  America/Chicago
  ICT  Wichita, Kansas  America/Chicago
  ILM  NWS Wilmington, NC  America/New_York
  ILN  Wilmington, OH  America/New_York
  ILX  Central Illinois  America/Chicago
  IND  Indianapolis, IN  America/Indiana/Indianapolis
  IWX  Northern Indiana  America/Indiana/Indianapolis
  JAN  Jackson, Mississippi  America/Chicago
  JAX  Jacksonville, FL  America/New_York
  JKL  Jackson, KY  America/New_York
  KEY  Key West, FL  America/New_York
  LBF  North Platte, NE  America/Chicago
  LCH  Lake Charles, LA  America/Chicago
  LIX  New Orleans/Baton Rouge  America/Chicago
  LKN  Elko, NV  America/Los_Angeles
  LMK  Louisville, KY  America/New_York
  LOT  Chicago, IL  America/Chicago
  LOX  Los Angeles, CA  America/Los_Angeles
  LSX  St. Louis, MO  America/Chicago
  LUB  Lubbock, TX  America/Chicago
  LWX  Baltimore/Washington  America/New_York
  LZK  Little Rock, AR  America/Chicago
  MAF  Midland/Odessa  America/Chicago
  MEG  Memphis, TN  America/Chicago
  MFL  Miami - South Florida  America/New_York
  MFR  Medford, OR  America/Los_Angeles
  MHX  Newport/Morehead City, NC  America/New_York
  MKX  Milwaukee/Sullivan, WI  America/Chicago
  MLB  Melbourne, FL  America/New_York
  MOB  Mobile/Pensacola  America/Chicago
  MPX  Twin Cities, MN  America/Chicago
  MQT  Marquette, MI  America/Detroit
  MRX  Morristown, TN  America/New_York
  MSO  Missoula, MT  America/Denver
  MTR  San Francisco Bay Area, CA  America/Los_Angeles
  OAX  Omaha/Valley, NE  America/Chicago
  OHX  Nashville, TN  America/Chicago
  OKX  New York, NY  America/New_York
  OTX  Spokane, WA  America/Los_Angeles
  OUN  Norman, OK  America/Chicago
  PAH  Paducah, KY  America/Chicago
  PBZ  Pittsburgh, PA  America/New_York
  PDT  Pendleton, OR  America/Los_Angeles
  PHI  Philadelphia/Mt Holly  America/New_York
  PIH  Pocatello, ID  America/Boise
  PPG  WSO Pago Pago  Pacific/Pago_Pago
  PQR  Portland, OR  America/Los_Angeles
  PSR  NWS Phoenix  America/Phoenix
  PUB  Pueblo, CO  America/Denver
  RAH  Raleigh, NC  America/New_York
  REV  Reno, NV  America/Los_Angeles
  RIW  Western and Central Wyoming  America/Denver
  RLX  Charleston, WV  America/New_York
  RNK  Blacksburg, VA  America/New_York
  SEW  Seattle/Tacoma, WA  America/Los_Angeles
  SGF  Springfield, MO  America/Chicago
  SGX  San Diego, CA  America/Los_Angeles
  SHV  Shreveport, LA  America/Chicago
  SJT  San Angelo, TX  America/Chicago
  SJU  San Juan, PR  America/Puerto_Rico
  SLC  Salt Lake City, UT  America/Denver
  STO  Sacramento, CA  America/Los_Angeles
  TAE  Tallahassee, FL  America/New_York
  TBW  Tampa Bay Area, FL  America/New_York
  TFX  Great Falls, MT  America/Denver
  TOP  Topeka, KS  America/Chicago
  TSA  Tulsa, OK  America/Chicago
  TWC  NWS Tucson Arizona  America/Phoenix
  UNR  Rapid City, SD  America/Denver
  VEF  Las Vegas, NV  America/Los_Angeles
